### DISCUSSION TITLE
Just released: JVM Code Base Exploration and Visualization

### DISCUSSION THREAD

**Discussion by dpolivaev** [freeplane/freeplane#1584] [Announcements]
As part of my daily work for my employer, [Unite](https://unite.eu/), I have developed a working prototype of a tool for analyzing, visualizing, and exploring JVM code bases. This tool utilizes the well-known architecture test library [ArchUnit](https://www.archunit.org/) for reading JVM class files and integrates with Freeplane to display these files in a mind map format.

The company has granted permission to share my implementation as free software under the same license as Freeplane. This is a significant development, as there are no comparable free or open-source tools available to my knowledge.

Members of the Freeplane community have the unique opportunity to be among the first to test and use this tool. It is available in the Freeplane preview version 1.11.9_01, accessible through Maps->Modes->Code Explorer.

I have also recorded a screencast to demonstrate how the tool can be used. Currently, it's an unlisted video on YouTube, but it will become public with the release of the new stable Freeplane version. If you're interested, you can view it at https://youtu.be/zqCt20rukrU. Feel free to share this link with anyone who might find it useful.

**Comment by EdoFro** [comment]
Wow! It looks impressive!
It doesn't seem to be something for the normal user, but I imagine there must be a niche that will know how to take great advantage of this.

Congratulations!

**Comment by sifran-github** [comment]
Impressive. Looks good !

**Comment by EdoFro** [comment]
I played with the new "Code Explorer" mode. It is really interesting.
I tried to follow the video but I was not able to understand it all.

Congrats!!

**Comment by dpolivaev** [comment]
There is an update with some bug fixes and a new version of the intro video https://youtu.be/zqCt20rukrU where I changed the middle part explaining the dependencies panel. The old video is not available any longer.

**Comment by nnako** [comment]
Nice one! I like the design, even though (due to the lack of Java experiance) I don't get all the use cases.
